### PR TITLE
feat: adding new "team_settings" table to Hasura

### PR DESCRIPTION
**Pull request by RODO94**
## What does this PR do?

This PR focuses on the Hasura side of adding a new ``team_settings`` table to the database. In future, this will allow users to manage their own settings for their Team, and connect to elements of the API and Editor. 

## Where to find more info?

I have been documenting some of what I'm doing here: [team_settings notion](https://www.notion.so/opensystemslab/Team-Settings-Work-9836b2122f6f43908cedf0bf9cd89953)

## Notable exclusions

I am yet to add in space for the computed field ``boundary__bbox`` but I will look to add this in during the current sprint.

``settings`` and ``notify_personalisation`` have no been dropped from the ``TEAMS`` table - this will be done in clean up near the end of the sprint.

### CODE REVIEW

**Review by DafyddLlyr — CHANGES_REQUESTED** [review]
A few comments to take a look at 👍 

It's a good decision not to remove the columns from the `team` table here - we can come back and tidy this up once the feature is complete.

I think there's a bit of a chicken-egg situation with the sync scripts which is causing CI to fail. Without the `team_settings` table on production, the sync script can't read from it. What this means is that we'll need to - 
 - Deploy the changes creating the table to staging and then to production
 - Then open a PR with the sync script updates which we'll also progress forward
 
You could just comment out the file in `main.sql` and we'll just comment it back in later 👍 
 
It's a known issue and not unique to this PR - we're going to need a better solution eventually here. Open to ideas!

File: hasura.planx.uk/metadata/tables.yaml [hunk 1]
```diff
@@ -1649,6 +1649,115 @@
     - role: platformAdmin
       permission:
         filter: {}
+- table:
```
[the remark this answers is not in the document]
Comment (DafyddLlyr) [reply hasura.planx.uk/metadata/tables.yaml:1652]: Yep, that's the one! Looks correct to me - you can check that the filter matches `team_themes` (and I think there may be even a way of copying directly).

File: hasura.planx.uk/metadata/tables.yaml [hunk 2]
```diff
@@ -1755,7 +1755,14 @@
...
           - reference_code
-        filter: {}
+        filter:
+          team:
+            members:
+              _and:
+                - user_id:
+                    _eq: x-hasura-user-id
+                - role:
+                    _eq: teamEditor
```
Comment (DafyddLlyr) [inline hasura.planx.uk/metadata/tables.yaml:1765]: 👍

**Review by DafyddLlyr — APPROVED** [review]

**Comment by RODO94** [comment]
[Create `team_settings` table](https://trello.com/c/ttAMDUjk/2511-create-teamsettings-table)

**Pull request merged**